### DISCUSSION TITLE
Missing Licensing for `chaostoolkit-` extentions

### DISCUSSION THREAD

**Discussion by cdsre** [chaostoolkit/chaostoolkit#299] [General]
There was a recent [issue](https://github.com/chaostoolkit/chaostoolkit/issues/286) that is caused as a result of the `chaostoolkit-addons` package not having a licence type. According to the `contributions` pages, all projects in the toolkit org should have an APL 2.0 licence. This is raised as a discussion since I am not sure how many projects might not have a licence.

It also opens a wider question of do we require extensions to have a licence type? If not then we should default this with something when not found. If it should have one then we need to filter out extensions that don't define one. Lastly if they must have one does it need to be APL 2.0

**Comment by Lawouach** [comment]
Hi Chris,

Thanks for raising the point. My thoughts are:

* We should not require any particular license
* However, we may suggest that we default to APL 2. Notably when using the generator https://github.com/chaostoolkit/chaostoolkit-extension-template
* We should indeed mark "Unknown" or something when the license is missing

**Comment by cdsre** [comment]
I think this is a sensible approach. I will raise an issue and submit an MR to default the licence type to Unknown when `chaostoolkit info extensions` is called.

Do we also want to make a cookie cutter template for the extensions template rather than people having to fork, or clone it then change the origin repo etc?

**Comment by Lawouach** [comment]
We had started with cookiecutter. I'm not sure it's as popular as it used to be so I don't have much perspective on this. Always remember that everything we had is maintainance for later ;)
